### DISCUSSION TITLE
PlotIPDF but with interpolated contours, does not work with contourf as option

### DISCUSSION THREAD

**Discussion by ZhangxiJesseFeng** [mtex-toolbox/mtex#1775] [Ask Anything]
I see from the documentation that plotIPDF has option "contourf". However, using it returns a blank plot.

I have:
`plotIPDF(ori0,zvector,'property',vmS/mean(vmS),'antipodal','minmax','on','MarkerSize',6);`

ori0 is a 400x1 orientation set, created by reading in 400 sets of Euler angles and with orientation.byEuler.
vmS/mean(vmS) is a 400x1 data set of von mises stress corresponding to each of the 400 orientations.

The above command plots the Von Mises stress as points in an inverse pole figure. When antipodal is changed to contourf, the result is a blank figure instead. The expectation is instead of points, the entire IPF is filled and color is set according to interpolated color values between two points.

What is the best way to plot an interpolated contour plot given the data that I have? Is it to take each Euler angle, calculate rotation matrix, and create vedctor3d?

Jesse

**Comment by lfgmorales** [comment]
Hi

Not sure if I understood this correctly, but the options ‚‘antipodal‘ and ‚contourf‘ are not interchangeable, you can use both together. That said, a proper way to calculate this would be calcDensity (ori) and than plot

Cheers

Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 21 Mar 2023, at 05:26, ZhangxiJesseFeng ***@***.***> wrote:

﻿

I see from the documentation that plotIPDF has option "contourf". However, using it returns a blank plot.

I have:
plotIPDF(ori0,zvector,'property',vmS/mean(vmS),'antipodal','minmax','on','MarkerSize',6);

ori0 is a 400x1 orientation set, created by reading in 400 sets of Euler angles and with orientation.byEuler.
vmS/mean(vmS) is a 400x1 data set of von mises stress corresponding to each of the 400 orientations.

The above command plots the Von Mises stress as points in an inverse pole figure. When antipodal is changed to contourf, the result is a blank figure instead. The expectation is instead of points, the entire IPF is filled and color is set according to interpolated color values between two points.

What is the best way to plot an interpolated contour plot given the data that I have? Is it to take each Euler angle, calculate rotation matrix, and create vedctor3d?

Jesse

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1775>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN2OFFAIEFA5B6PSU5LW5EUYHANCNFSM6AAAAAAWB5TOMI>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1775", "url": "https://github.com/mtex-toolbox/mtex/discussions/1775", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by ZhangxiJesseFeng** [reply]
> Hi Not sure if I understood this correctly, but the options ‚‘antipodal‘ and ‚contourf‘ are not interchangeable, you can use both together. That said, a proper way to calculate this would be calcDensity (ori) and than plot Cheers Luiz Luiz F. G. Morales ScopeM - ETH Zürich Sent from iPhone On 21 Mar 2023, at 05:26, ZhangxiJesseFeng ***@***.***> wrote: ﻿ I see from the documentation that plotIPDF has option "contourf". However, using it returns a blank plot. I have: plotIPDF(ori0,zvector,'property',vmS/mean(vmS),'antipodal','minmax','on','MarkerSize',6); ori0 is a 400x1 orientation set, created by reading in 400 sets of Euler angles and with orientation.byEuler. vmS/mean(vmS) is a 400x1 data set of von mises stress corresponding to each of the 400 orientations. The above command plots the Von Mises stress as points in an inverse pole figure. When antipodal is changed to contourf, the result is a blank figure instead. The expectation is instead of points, the entire IPF is filled and color is set according to interpolated color values between two points. What is the best way to plot an interpolated contour plot given the data that I have? Is it to take each Euler angle, calculate rotation matrix, and create vedctor3d? Jesse — Reply to this email directly, view it on GitHub<#1775>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN2OFFAIEFA5B6PSU5LW5EUYHANCNFSM6AAAAAAWB5TOMI>. You are receiving this because you are subscribed to this thread.Message ID: ***@***.***> [ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "#1775", "url": "#1775", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

Hi Luiz,

Thanks for responding! I see that I can calculate an ODF = calcDensity(ori0). Then I can use:
`plotIPDF(odf,zvector,'property',vmS/mean(vmS),'minmax','on','MarkerSize',12);`
But I am not able to use "property" to overwite the value on the IPF. The ODF gives max/min of 1.1/0.9, instead of the expected 1.3/0.72 from the vmS data, even when I included 'property' option.

I can use the plot function, but I am not sure about the correct syntax. When I do plot(odf,...) it gives a series of odf plots and not as an inverse pole figure.

The goal is to plot the vmS data on top of the inverse pole figure, and instead of having the IPF be individual points plotted with markers, it would be plotted like a contour plot.

Thanks!

Jesse
